[shadhankkk] iP

shadhankkk opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

shadhankkk committed:
Completed Level-0

shadhankkk committed:
Completed Level-1

shadhankkk committed:
Completed Level-2

shadhankkk committed:
Completed Level-3

shadhankkk committed:
Completed Level-4

shadhankkk committed:
Completed A-TextUiTesting

shadhankkk committed:
Completed Level-5

shadhankkk committed:
Completed Level-6

shadhankkk committed:
Level-7

shadhankkk committed:
Completed Level-8

shadhankkk committed:
Completed A-MoreOOP

shadhankkk committed:
Completed A-Packages

shadhankkk committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

shadhankkk committed:
Completed A-JUnit

shadhankkk committed:
Completed A-Jar

shadhankkk committed:
Completed A-JavaDoc

shadhankkk committed:
Completed A-CodingStandard

shadhankkk committed:
Completed Level-9

shadhankkk committed:
Completed Week 3 iP

weiseng18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,897 @@
+        /**
+         * Returns void, just prints a horizontal line
+         */
+        void printHorizontalLine() {
```
method names follow convention (start with verb, written in camelCase) and are really organized! great work on that

weiseng18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,897 @@
+            int len = userInput.length();
+            int startIndex = ptr;
+
+            while(ptr < len) {
```
The recommended Java coding standard requires a space after reserved keywords such as `while`. But you'll probably figure this out after adding checkstyle in Week 4, so you can fix it then!

weiseng18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,897 @@
+
+            ui.printHorizontalLine();
+
+            for(int i=0; i<numberOfMessages; i++) {
```
Same here for space after `for`

weiseng18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,897 @@
+         * in our array of tasks
+         */
+        void markTask(int rank) {
+            if(rank < 1 || rank > tasks.size()) {
```
Same here for space after `if`, etc.

weiseng18 reviewed (commented):
overall very organized code!

there are some minor coding style issues that you'll probably notice after adding checkstyle in Week 4, so you can fix it once you've worked on that!

you might want to add `.class` and `.jar` to `.gitignore` since we don't want to commit binaries

yooplo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,897 @@
+        /**
+         * Returns void, just prints the opening text
+         */
+        void printOpening() {
```
Looks good!I don't know if adding access modifiers to the methods would make it better though.

yooplo reviewed (commented):
Overall, well done!

sandyk0105 commented on src/main/java/duke/Duke.java:
```diff
@@ -63,20 +96,35 @@ void printTaskDeleted(Task task, int len) {
             task.print();
             System.out.println(horizontalLine);
         }
 
```
Ui class has all needed methods, using abstraction very very well.

shadhankkk committed:
Completed Level-10

shadhankkk committed:
Add assert statements to improve code safety

Some parts of the code assume certain conditions hold, but these assumptions are not explicitly enforced. This could lead to errors or unexpected behavior during execution if those assumptions are violated.

Without enforcing these conditions, the code may fail silently, which could make debugging more difficult and reduce code reliability.

Let's add assert statements to verify critical conditions at runtime. This ensures that assumptions are checked and provides immediate feedback if they are violated.

Using assert is appropriate here because it adds minimal overhead and is disabled in production, keeping performance unaffected.

shadhankkk committed:
Remove dead code from the program

The program contains blocks of code that are no longer executed or necessary. These portions of the codebase add unnecessary complexity and make it harder to maintain and read the code.

Dead code can confuse developers and lead to misunderstandings about the program's behavior. It's important to keep the codebase clean and focused only on what is required.

Let's remove the dead code to simplify the codebase, reduce maintenance overhead, and improve overall readability.

Removing dead code helps avoid future bugs and keeps the project lean.

shadhankkk committed:
do A-UserGuide

shadhankkk committed:
Rename jeff_chatbot_ss.png to Ui.png

shadhankkk committed:
Rename jeff_chatbot_ss.png to Ui.png

shadhankkk committed:
Merge branch 'master' of https://github.com/shadhankkk/ip

shadhankkk committed:
Release v0.3

shadhankkk committed:
Add User Guide

shadhankkk committed:
Fix README.md broken image link

shadhankkk committed:
v0.6

shadhankkk committed:
Add release v0.6

shadhankkk committed:
Add README.md

shadhankkk committed:
Update opening text

shadhankkk committed:
Remove dead code

shadhankkk committed:
Update closing text

shadhankkk committed:
Remove unnecessary blank lines

shadhankkk committed:
Fix ip peer review bugs

shadhankkk committed:
Remove dead code

shadhankkk committed:
Update out-of-bounds error handling
